wallet/src/repository/wallet/mod.rs

```diff
@@ -496,6 +504,24 @@ where
                 _ => continue,
             };
 
+            // Check if tally txn corresponds to a wallet sent data request
+            if let types::Transaction::Tally(tally) = &txn {
+                // There is a DR transaction persisted in database whose tally was found
+                if self
+                    .db
+                    .get::<_, model::Path>(&keys::transactions_index(tally.dr_pointer.as_ref()))
+                    .is_ok()
```

Comment:
I think `self.db.get` returns `Result<Option<T>>`, so it will always be ok. To check if some item exists in the database you should check if the result is `Ok(Some(_))`.

Reply:
Or maybe there is a `.contains`

Reply:
I think there is no `contains`, but it should be really easy to implement. :)

Reply:
Not today